harness/determined/cli/sso.py

```diff
@@ -73,17 +77,33 @@ def sso(parsed_args: Namespace) -> None:
         matched_provider = matching_providers[0]
 
     sso_url = matched_provider["sso_url"] + "?relayState=cli"
-    webbrowser.open(sso_url)
-    print(
-        "Your browser should open and prompt you to sign on;"
-        " if it did not, please visit {}".format(sso_url)
-    )
-
-    with HTTPServer(
-        ("localhost", CLI_REDIRECT_PORT),
-        make_handler(parsed_args.master, lambda code: sys.exit(code)),
-    ) as httpd:
-        httpd.serve_forever()
+
+    if not os.environ.get("DISPLAY"):
```

Comment:
DISPLAY is x11-specific, right?  what about macos, windows, and wayland users?

Reply:
Ahh, very true. This will trigger in all of those cases 🤦

Let me see if I can figure out if the call to `webbrowser.open()` fails and just make the condition based on that?

Reply:
It looks like I can just use this `webbrowser.open()` return here. I made that the condition along with adding a `--headless` flag in the event that, for some reason, `webbrowser.open()` returns true, but we still can't access localhost.